### ISSUE TITLE
how to use multiple cpu to speed up trainning process

### ISSUE THREAD

**Issue by wanghaisheng** [mittagessen/kraken#17]
can your library support enabling multiple cpu to speed up trainning process?
if so ,how

**Comment by mittagessen** [comment]
Currently, kraken uses mainly the same code as ocropus so it isn't possible to accelerate training using multiple CPUs or GPU resources. In fact, doing so would require a complete reimplementation of the network architecture/training metric for another library. I've been doing some half-hearted work in that direction but there hasn't come anything off it.

**Comment by wanghaisheng** [comment]
will you use  tensorflow or something like that ?

**Comment by mittagessen** [comment]
I've evaluated theano or rather some frameworks based on it and it has some non-features making it undesirable to use, chiefly resource-intensive compilation that can't be offloaded/cached and clunky serialization. Recompiling/-linking the model took longer than the actual OCR for 10 pages which I believe to be unacceptable now that reinstantiating serialized models runs in <0.1s.

The last time I had a look at tensorflow its support for recurrent neural nets was weak, requiring a number of dirty hacks but it's entirely possible that this has changed by now. 

So yes I will use something like tensorflow or theano, especially as I'd like not to maintain another single-use-case library and experiment with different architectures in the future, but haven't found one fitting my needs. I'd be glad for suggestions though.

**Comment by zuphilip** [comment]
FYI: @tmbdev is "integrating CLSTM into TensorFlow, which gives you several options for parallel and GPU training of LSTM and image processing" https://github.com/tmbdev/ocropy/issues/80#issuecomment-176442987

**Comment by wanghaisheng** [comment]
have you ever looked at torch7 or mxnet  both seems perfect
